com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeGIBaking.cs

```diff
@@ -566,26 +580,38 @@ static void OnAdditionalProbesBakeCompleted()
                 data2Asset[data] = ProbeVolumeAsset.CreateAsset(data.gameObject.scene);
             }
 
-            // Put cells into the respective assets
+            // Allocate cells to the respective assets
+            var asset2BakingCells = new Dictionary<ProbeVolumeAsset, List<BakingCell>>();
             foreach (var cell in m_BakedCells.Values)
             {
                 foreach (var scene in m_BakingBatch.cellIndex2SceneReferences[cell.index])
                 {
                     // This scene has a reference volume authoring component in it?
-                    ProbeVolumePerSceneData data = null;
-                    if (scene2Data.TryGetValue(scene, out data))
+                    if (scene2Data.TryGetValue(scene, out var data))
                     {
                         var asset = data2Asset[data];
-                        asset.cells.Add(cell);
+
+                        if (!asset2BakingCells.TryGetValue(asset, out var bakingCellsList))
+                            bakingCellsList = asset2BakingCells[asset] = new();
+
+                        bakingCellsList.Add(cell);
+
                         var profile = probeRefVolume.sceneData.GetProfileForScene(scene);
                         asset.StoreProfileData(profile);
-                        Debug.Assert(profile != null);
```

Comment:
Any reason why removing this assert? Admittedly this should be before the StoreProfileData and might be outdated, but I wonder if there is any specific reason you removed it.

Reply:
I removed it because StoreProfileData accesses fields of profile, so it can never trigger, it would always throw a NRE before getting to it.